packages/components/src/components/segmented-button/segmented-button.tsx

```diff
@@ -79,22 +79,26 @@ export class SegmentedButton {
   showHelperText = false;
   @Listen('scaleClick')
   scaleClickHandler(ev: { detail: { id: string; selected: boolean } }) {
-    let tempState: SegmentStatus[];
+    if (this.status.length === 0) {
+      return;
+    }
+    let tempState = this.getAllSegments().map((segment) => {
+      return {
+        id: segment.getAttribute('segment-id') || segment.segmentId,
+        selected:
+          segment.hasAttribute('selected') && segment.selected ? true : false,
+      };
+    });
     if (!this.multiSelect) {
       if (!ev.detail.selected) {
-        tempState = this.status.map((obj) =>
+        tempState = tempState.map((obj) =>
           ev.detail.id === obj.id ? ev.detail : { ...obj }
         );
-        /* clicked button has now selected state */
       } else {
-        tempState = this.status.map((obj) =>
+        tempState = tempState.map((obj) =>
           ev.detail.id === obj.id ? ev.detail : { ...obj, selected: false }
         );
       }
-    } else {
-      tempState = this.status.map((obj) =>
-        ev.detail.id === obj.id ? ev.detail : { ...obj }
-      );
     }
     this.setState(tempState);
```

Comment:
The `scaleClickHandler`  creates the `SegmentStatus` array for the `setStatus`  to set the state and carry out side effects.
 in the current implementation, we create this from manipulating the current state and adapting the value that has changes from `event.detail`. But in your implementation, the object is created each time from scratch, which would make the second part of checking and changing of `tempState` based on the event details not relevant.  We have to either go with the previous logic or with your logic we have to remove the extra assignments of selected attributes